### PR TITLE
EP-513-4: Remove Datalake Events And Worker

### PR DESCRIPTION
**Pull request by sunday-okpoluaefe**
# 📲 What

 Android: Remove Datalake Events And Worker

# 🤔 Why

- With the completion of the Segment events, we need to remove all of the code used for sending Datalake/Koala events as they are no longer needed.

# 🛠 How

- Removed LakeTrackingClient from the application
- Remove LAKE from the TrackingClientType

- Removed KoalaEvent enum and the affecting functions from the app

- Removed All unused functions from the AnalyticsEvents class

- Removed Any additional references to the DataLake are removed from the app

- Removed LakeWorker needs to be removed

- Deleted the provider for the Lake worker, and the provider for the retrofit service from the AppModule

- Updated Tests to reflect the removed changes

# 👀 See

Trello, screenshots, external resources?

| Before 🐛 | After 🦋 |
| --- | --- |
|  |  |

# 📋 QA

Instructions for anyone to be able to QA this work.

# Story 📖

https://kickstarter.atlassian.net/browse/EP-513
https://kickstarter.atlassian.net/browse/EP-514

### CODE REVIEW

File: app/src/main/java/com/kickstarter/libs/FragmentViewModel.java [hunk 1]
```diff
@@ -25,11 +25,11 @@
   private final PublishSubject<ActivityResult> activityResult = PublishSubject.create();
   private final PublishSubject<Bundle> arguments = PublishSubject.create();
   protected final PublishSubject<Boolean> isExpanded = PublishSubject.create();
-  protected final AnalyticEvents lake;
+  protected final AnalyticEvents analyticEvents;
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/libs/FragmentViewModel.java:28]: Thank you for the re-naming  👍

File: app/src/main/java/com/kickstarter/libs/PushNotifications.java [hunk 2]
```diff
@@ -335,15 +339,16 @@ private void displayNotificationFromUpdateActivity(final @NonNull PushNotificati
     return taskStackBuilder.getPendingIntent(envelope.signature(), PendingIntent.FLAG_UPDATE_CURRENT);
   }
 
-  private @NonNull PendingIntent messageThreadIntent(final @NonNull PushNotificationEnvelope envelope,
+  @VisibleForTesting
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/libs/PushNotifications.java:342]: 👍

File: app/src/test/java/com/kickstarter/KSRobolectrictTestCase.kt [hunk 3]
```diff
@@ -84,15 +82,15 @@ abstract class KSRobolectricTestCase : TestCase() {
...
-    private fun lakeTrackingClient(mockCurrentConfig: MockCurrentConfig, experimentsClientType: MockExperimentsClientType): MockTrackingClient {
-        lakeTest = TestSubscriber()
-        val lakeTrackingClient = MockTrackingClient(
-            MockCurrentUser(),
-            mockCurrentConfig, TrackingClientType.Type.LAKE, experimentsClientType
-        )
-        lakeTrackingClient.eventNames.subscribe(lakeTest)
-        return lakeTrackingClient
-    }
+//    private fun lakeTrackingClient(mockCurrentConfig: MockCurrentConfig, experimentsClientType: MockExperimentsClientType): MockTrackingClient {
```
Comment (Arkariang) [inline app/src/test/java/com/kickstarter/KSRobolectrictTestCase.kt:85]: Can be removed

File: app/src/test/java/com/kickstarter/libs/PushNotificationsTest.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import android.app.Application
+import androidx.test.core.app.ApplicationProvider
+import com.kickstarter.KSRobolectricTestCase
+import com.kickstarter.mock.factories.MessageThreadEnvelopeFactory
+import com.kickstarter.mock.factories.PushNotificationEnvelopeFactory
+import com.kickstarter.ui.IntentKey
+import org.junit.Test
+
+class PushNotificationsTest : KSRobolectricTestCase() {
+    private val application: Application = ApplicationProvider.getApplicationContext()
```
Comment (Arkariang) [inline app/src/test/java/com/kickstarter/libs/PushNotificationsTest.kt:12]: `KSRobolectricTestCase` has a protected method `application()` to fetch the context Application, take a look at `SegmentTest.kt:73`

**Review by Arkariang — APPROVED** [review]
Just two small comments left. Apart from that amazing job on this guys @sunday-okpoluaefe  @hadia 👏 🎆 🚀

File: app/src/test/java/com/kickstarter/libs/PushNotificationsTest.kt (hunk 4 above)
Comment (hadia) [reply app/src/test/java/com/kickstarter/libs/PushNotificationsTest.kt]: 👍🏻

File: app/src/test/java/com/kickstarter/KSRobolectrictTestCase.kt (hunk 3 above)
Comment (hadia) [reply app/src/test/java/com/kickstarter/KSRobolectrictTestCase.kt]: 👍🏻

File: app/src/main/java/com/kickstarter/libs/FragmentViewModel.java (hunk 1 above)
Comment (hadia) [reply app/src/main/java/com/kickstarter/libs/FragmentViewModel.java:28]: 😊😊

**Pull request merged**